### PR TITLE
Try to ingest SLSA

### PR DESCRIPTION
**Pull request by mihaimaruseac**
This is already quite large so I'll definitely split it further.

### CODE REVIEW

File: pkg/assembler/graphql/schema/hasSLSA.graphql [hunk 1]
```diff
@@ -15,85 +15,114 @@
...
   key: String!
   value: String!
 }
 
-"""
-HasSLSASpec allows filtering the list of HasSLSA to return.
-"""
+"HasSLSASpec allows filtering the list of HasSLSA to return."
 input HasSLSASpec {
   package: PkgSpec
```
Comment (pxp928) [inline pkg/assembler/graphql/schema/hasSLSA.graphql:122]: can this all be replaced by `PackageSourceOrArtifactSpec`

File: pkg/assembler/backends/neo4j/hasSLSA.go [hunk 2]
```diff
@@ -488,23 +489,20 @@ func (c *neo4jClient) HasSlsa(ctx context.Context, hasSLSASpec *model.HasSLSASpe
...
-		invalidSubject = true
+	if hasSLSASpec.Artifact != nil {
+		subjectsDefined = subjectsDefined + 1
 	}
-	if hasSLSASpec.Source != nil && hasSLSASpec.Artifact != nil {
-		invalidSubject = true
-	}
-	if invalidSubject {
-		return gqlerror.Errorf("cannot specify more than one subject for CertifyBad query")
+	if subjectsDefined > 1 {
```
Comment (lumjjb) [inline pkg/assembler/backends/neo4j/hasSLSA.go:504]: 👍

**Review by lumjjb — APPROVED** [review]
lgtm

File: pkg/assembler/graphql/schema/hasSLSA.graphql (hunk 1 above)
Comment (mihaimaruseac) [reply pkg/assembler/graphql/schema/hasSLSA.graphql:122]: Right, but that would require more changes on the query side, so I was punting on it. However, I already changed a lot, so will do

**Pull request closed without merging**